### PR TITLE
Add Terraform configuration for archive

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: terraform/domain.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
+resource "aws_route53_zone" "dandi" {
+  name = "dandiarchive.org"
+}
+
+resource "aws_route53_record" "apex" {
+  zone_id = aws_route53_zone.dandi.zone_id
+  name    = "" # apex
+  type    = "A"
+  ttl     = "300"
+  records = ["3.12.21.28"]
```
Comment (brianhelba) [inline terraform/domain.tf:10]: Redirector?

File: terraform/publish.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    DJANGO_DANDI_DANDISETS_BUCKET_NAME = aws_s3_bucket.publish_dandisets.id
+    DJANGO_DANDI_GIRDER_API_URL  = "https://girder.dandiarchive.org/api/v1"
+    DJANGO_SENTRY_DSN = "https://4bd48b5174ea4b42a130e63ebe3d60d2@o308436.ingest.sentry.io/5266078"
+  }
+  additional_sensitive_django_vars = {
+    DJANGO_DANDI_GIRDER_API_KEY = var.girder_api_key
+  }
+}
+
+resource "aws_s3_bucket" "publish_dandisets" {
```
Comment (brianhelba) [inline terraform/publish.tf:27]: Only for testing.

File: terraform/redirector.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+resource "aws_route53_record" "redirector" {
+  zone_id = aws_route53_zone.dandi.zone_id
+  name    = "" # apex
+  type    = "A"
+  ttl     = "300"
+  records = ["3.12.21.28"]
```
Comment (yarikoptic) [inline terraform/redirector.tf:6]: could it be a name? (`dandiarchive.org`)

File: terraform/girder.tf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  source  = "girder/girder/aws//modules/server"
+  version = "0.6.0"
+
+  project_slug      = "dandi-girder"
+  ssh_public_key    = data.local_file.ssh_public_key.content
+  ec2_volume_size = 100
+}
+
+module "girder_smtp" {
+  source  = "girder/girder/aws//modules/smtp"
```
Comment (yarikoptic) [inline terraform/girder.tf:23]: just curious, is there some added semantic behind `//`?

Comment (brianhelba) [reply terraform/girder.tf:23]: Yes, it's Terraform Registry's syntax for submodules: https://registry.terraform.io/modules/girder/girder/aws/0.7.0/submodules/smtp

File: terraform/redirector.tf (hunk 3 above)
Comment (brianhelba) [reply terraform/redirector.tf:6]: The root domain name is specified by the `aws_route53_zone.dandi` resource. This is a subdomain record. The `name` for an apex subdomain is the empty string.

The `aws_route53_record` resource allows you to provide a FQDN as the `name`, but while that's more explicit, it's also more fragile if the root domain ever changes or the definitions are reused with a different root domain. Once a developer is familiar with Terraform and our infrastructure, I don't think it's confusing.

**Pull request merged**